jcl/src/java.base/share/classes/jdk/internal/foreign/abi/UpcallLinker.java

```diff
@@ -113,5 +121,22 @@ public static MemorySegment make(MethodHandle target, MethodType mt, FunctionDes
 	public static UpcallStubFactory makeFactory(MethodType targetType, ABIDescriptor abi, CallingSequence callingSequence) {
 		throw new UnsupportedOperationException();
 	}
+
+	/**
+	 * The method constructs a factory instance (introduced in JDK21 to support the
+	 * sharing of function descriptor) which wraps up the existing upcall specific
+	 * code implemented in OpenJ9 to enable the callback in OpenJDK.
+	 *
+	 * @param methodType The MethodType of the upcall method handle
+	 * @param descriptor The FunctionDescriptor of the upcall method handle
+	 * @return a factory instance that wraps up the upcall specific code
+	 */
+	public static UpcallStubFactory makeFactory(MethodType methodType, FunctionDescriptor descriptor) {
+		UpcallStubFactory factory = (target, session) -> {
```

Comment:
Why the local? Why not simply a return statement?

Reply:
Agreed and fixed.